src/Controller/Controller_Pdf_Queue.php

```diff
@@ -278,6 +279,27 @@ protected function get_queue_tasks( $entry, $form, $notifications = [] ) {
 
 		$queue_data = apply_filters( 'gfpdf_queue_initialise', [], $entry, $form );
 
+		/**
+		 * Check if Asynchronous notification is enabled for this form.
+		 * https://docs.gravityforms.com/gform_is_asynchronous_notifications_enabled/
+		 */
+		$is_asynchronous = gf_apply_filters(
+			[
+				'gform_is_asynchronous_notifications_enabled',
+				$form['id'],
+			],
+			false,
+			$notifications['event'],
+			$notifications['id'],
+			$form,
+			$entry,
+			[]
+		);
+
+		if ( $is_asynchronous ) {
+			return $queue_data;
+		}
+
 		/* Queue up the PDF generation callback */
 		if ( count( $pdfs ) > 0 ) {
 			$notifications = ( count( $notifications ) > 0 ) ? $notifications : $this->get_active_notifications( $form, $entry );
```

Comment:
Move to here: `array_filter( $notifications, function( $notifications ) { return ! apply_filters( 'gform_is_asynchronous_notifications_enabled' ... ) }`